net/socket/socket.h

```diff
@@ -94,18 +94,30 @@
 /* Macro to set socket errors */
 
 #ifdef CONFIG_NET_SOCKOPTS
+#  define _SO_CONN_SETERRNO(c,e) \
+    do \
+      { \
+        (c)->s_error = (int16_t)e; \
+        set_errno(e); \
+      } \
+    while (0)
+
 #  define _SO_SETERRNO(s,e) \
     do \
       { \
         if (s != NULL && (s)->s_conn != NULL) \
           { \
             FAR struct socket_conn_s *_conn = (s)->s_conn; \
-            _conn->s_error = (int16_t)e; \
+            _SO_CONN_SETERRNO(_conn, e); \
```

Comment:
_SO_CONN_SETERRNO((s)->s_conn, e);
and remove line 109

Reply:
Done